public/app/core/services/NewFrontendAssetsChecker.ts

```diff
@@ -0,0 +1,51 @@
+import { getBackendSrv, getGrafanaLiveSrv } from '@grafana/runtime';
+
+export class NewFrontendAssetsChecker {
+  private hasUpdates = false;
+  private previous = '';
+  private interval = 1000 * 60 * 4; // force check after 5 mins
+  private checked = Date.now() - this.interval / 5;
+
+  public start() {
+    const startup = Date.now();
+    let prevState = false;
+
+    // Subscribe to live connection state changes and check for new assets when re-connected
+    const live = getGrafanaLiveSrv();
+    if (live) {
+      live.getConnectionState().subscribe((connected) => {
+        const elapsed = Date.now() - startup;
+        if (elapsed > 1000 && connected && connected !== prevState) {
```

Comment:
this will always be true anytime after the first second passed. I am not sure how often the subscribe callback executes, is this intended?

Reply:
@academo think this meant to ignore connection events that happen at app startup.

Reply:
The connectionState events happen when frontend connects to the server -- if the server stays online it is only at startup, but if it restarts, a new connection will be resetablished